src/main/java/Duke.java

```diff
@@ -1,10 +1,141 @@
+    
+        public String getStatusIcon() {
+            return (isDone ? "X" : " "); // mark done task with X
+        }
+    
+        public void markAsDone(){
+            this.isDone = true;
+        }
+
+        public void markAsNotDone(){
+            this.isDone = false;
+        }
+
+        public String getDescription(){
+            return this.description;
+
+        }
+
+        public String getDate(){
+            return this.taskDate;
+    }
+}
+
+
+    // Driver code
     public static void main(String[] args) {
+        // Greeting message
         String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
+                    + "|  _ \\ _   _| | _____ \n"
+                    + "| | | | | | | |/ / _ \\\n"
+                    + "| |_| | |_| |   <  __/\n"
+                    + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+        System.out.println("Hello! I'm Duke_HTT. \n What can I do for you?");
+        
+        // Take user inputt
+        Scanner scan = new Scanner ( System.in ); 
+        String inData = scan.nextLine();
+
+        // Processing input - 1st time
+        Vector<Task> tasks = new Vector<>();
+        
+        if (inData.equals("bye")){
+            System.out.println(" \n Bye. Hope to see you again soon!");
+        } else if (inData.equals("list")){
+            for (Task task:tasks){
+                System.out.println("Here are the tasks in your list: \n" 
+                                    + (tasks.indexOf(task) + 1) 
+                                    + "." + "[" + task.getStatusIcon() + "] " 
+                                    + task.getDescription());   
+            }
+        } else{
+            Task t = new Task(inData);
+            tasks.add(t);
+            System.out.println("added: " + t.getDescription());
+        }
+
+        // Processing input - loop
+        Boolean isBye = false;
+
+        while (!isBye) {
+
+            inData = scan.nextLine();
+            
+            if (inData.equals("bye")){
+                isBye = true;
+                System.out.println(" \n Bye. Hope to see you again soon!");
+            } else if (inData.equals("list")){
+                for (Task task:tasks){
+                    System.out.println("Here are the tasks in your list: \n" 
+                                        + (tasks.indexOf(task) + 1) 
+                                        + "[" + task.getDescription() + "]"
+                                        + "." + "[" + task.getStatusIcon() + "] " 
+                                        + task.getDescription());
+                }
+            } else if (inData.contains("mark") && !inData.contains("unmark")){
+                int inDataIndex = Integer.parseInt(inData.substring(5));
+                inDataIndex--;
```

Comment:
A comment here to explain the reason for -- would be helpful, especially in future reviews of the code, since a random -- might be a potential cause of bugs